fusequery/query/src/sql/plan_parser.rs

```diff
@@ -499,6 +496,25 @@ impl PlanParser {
         }
     }
 
+    /// Apply a having to the plan
+    fn having(
```

Comment:
Is it a good idea to consolidate `filter` and `having` ?

Reply:
For now, it is good to stand alone, for the future it is good to consolidate with filter, such as filter has an alias, etc.

Reply:
The disadvantage is that we need to make optimizers for `having`, and this patch doesn't add the `having` to optimizer.
Datafuse are functional first, this can be opened an new issue for optimizer.
FYI @tlightsky

Reply:
got it, I'll try to figure it out later